woocommerce/woocommerce-android
# Tracking receipts

Parent issue #4039 

Merge instructions
1. Make sure https://github.com/woocommerce/woocommerce-android/pull/4244 is merged [x]
2. Remove "do not merge" label
3. Merge this PR

This PR adds tracking to send/print receipt functionality.
receipt_email_tapped: The user tapped on the button to email a receipt.
receipt_email_failed: Email application not found in the system.

receipt_print_tapped: The user tapped on the button to print a receipt.
receipt_print_failed: Printing the receipt failed.
receipt_print_canceled: The user canceled printing the receipt.
receipt_print_success: The print job was passed to the OS.

Note 1:
The iOS app tracks also 
receipt_email_canceled: The user canceled sending the receipt by email.
receipt_email_success: The receipt was sent by email.
However, the Android OS doesn't inform the source app about results of implicit intents, so these events are not used on Android.

Note 2:
Tracking print result - PrintManager doesn't provide a public callback which would inform the caller about the result of a print job. The current status of a PrintJob is available on the PrintJob itself. We could query the PrintJob every few seconds to find out the current status and track the events. However, implementing it would not be trivial. Therefore my suggestion is to track the current status of the print job when the WCAndroid comes back to the foreground from the OS print UI. If the job is enqueued/started/completed/blocked we track `success` event, if it's failed we track `failed` event (even though the user can still retry), if it's canceled we track `cancelled` event. This logic should be pretty close to how the iOS app behaves.

To test:
Verify that the events are tracked - you can use "print to pdf" if you don't have a printer available. I'm not sure how to simulated "Print failed" state, but since it shares the same codepath with success/cancelled I don't think we need to test it.


Update release notes:
